language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala

```diff
@@ -155,21 +155,33 @@ object CodegenConfigReader {
   private def resultR[A](a: A): Result[A] =
     Right(a): Result[A]
 
-  implicit val decodePackageReference: KeyDecoder[PackageReference] =
-    new KeyDecoder[PackageReference] {
-      // TODO (MK) https://github.com/digital-asset/daml/issues/9934
-      // For now we only allow name-version pairs to match the compiler. Once the compiler
-      // accepts package ids we can allow for those here as well.
-      final def apply(key: String): Option[PackageReference] =
-        key.splitAt(key.lastIndexOf('-')) match {
-          case (rawPackageName, rawPackageVersion)
-              if rawPackageName.nonEmpty && rawPackageVersion.nonEmpty =>
-            for {
-              name <- PackageName.fromString(rawPackageName).toOption
-              version <- PackageVersion.fromString(rawPackageVersion.tail).toOption
-            } yield PackageReference.NameVersion(name, version)
-          case _ =>
-            None
-        }
+  private[conf] def splitNameAndVersion(
+      string: String,
+      separator: Char,
+  ): Option[(String, String)] = {
+    val separatorIndex = string.lastIndexOf(separator.toInt)
```

Comment:
why do you need a `toInt`? https://www.scala-lang.org/api/2.12.7/scala/collection/immutable/StringOps.html#lastIndexOf(elem:Char):Int seems to take a `Char`.

Reply:
You are pointing at 2.12.7 docs. Not sure whether that is present in the new collection API that has been rolled out as part of 2.13. If I don't turn it explicitly into an integer, I get a warning (which I regard as valid):

```
language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala:164: warning: implicit numeric widening
    val separatorIndex = string.lastIndexOf(separator)
                                            ^
error: No warnings can be incurred under -Werror.
```

as it tries to use [this](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#lastIndexOf(int)).